lib/types/local-date.js

```diff
@@ -26,58 +22,81 @@ var twentiethCentury = new Date('1899-12-31') - new Date('0000-01-01');
  * <p>
  *   This class does not store or represent a time or time-zone. Instead, it is a description of the date, as used for birthdays. It cannot represent an instant on the time-line without additional information such as an offset or time-zone.
  * </p>
- * @param {Number} year The year.
+ * <p>
+ *   Note that this type can represent dates in the range [-5877641-06-23; 5881580-07-17] while the ES5 date type can only represent values in the range of [-271821-04-20; 275760-09-13].
+ *   In the event that year, month, day parameters do not fall within the ES5 date range an Error will be thrown.  If you wish to represent a date outside of this range, pass a single
+ *   parameter indicating the days since epoch.  For example, -1 represents 1969-12-31.
+ * </p>
+ * @param {Number} year The year or days since epoch.  If days since epoch, month and day should not be provided.
  * @param {Number} month Between 1 and 12 inclusive.
  * @param {Number} day Between 1 and the number of days in the given month of the given year.
+ *
+ * @property {Date} date The date representation if falls within a range of an ES5 data type, otherwise an invalid date.
+ * @property {Number} value The number of days since epoch.
+ *
  * @constructor
  */
 function LocalDate(year, month, day) {
   //implementation detail: internally uses a UTC based date
   if (typeof year === 'number' && typeof month === 'number' && typeof day === 'number') {
-    this.value = new Date(Date.UTC(year, month - 1, day));
-    if (year >= 0 && year < 100) {
-      //If there is a 2 digit year, Date.UTC() assumes that is the 20th century
-      //Thanks ECMAScript!
-      this.value = new Date(this.value.getTime() - twentiethCentury);
+    //Use setUTCFullYear as if there is a 2 digit year, Date.UTC() assumes
+    //that is the 20th century.  Thanks ECMAScript!
+    var date = new Date();
+    date.setUTCHours(0, 0, 0, 0);
+    date.setUTCFullYear(year, month-1, day);
+    if(isNaN(date.getTime())) {
+      throw new Error(util.format('%d-%d-%d does not form a valid ES5 date!',
+        year, month, day));
     }
+
+    this.date = date;
+    this.value = Math.floor(this.date.getTime() / millisecondsPerDay);
   }
   else if (typeof month === 'undefined' && typeof day === 'undefined') {
     if (typeof year === 'number') {
-      //in milliseconds since unix epoch
-      this.value = new Date(year);
+      //in days since epoch.
+      if(year < -2147483648 || year > 2147483647) {
+        throw new Error('You must provide a valid value for days since epoch (-2147483648 <= value <= 2147483647).')
+      }
+      this.value = year;
```

Comment:
I would try to avoid exposing it for now and reduce the allocation whenever possible.
We could even make it not enumerable initially:

```javascript
Object.defineProperty(this, 'value', { enumerable: false, value: null });
```

But as long it is undocumented, it is fine by me.